x-pack/filebeat/input/cometd/input.go

```diff
@@ -37,44 +44,89 @@ func (in *cometdInput) Run() {
 			defer in.workerWg.Done()
 			defer in.workerCancel()
 			in.b = bay.Bayeux{}
-			in.creds, err = bay.GetSalesforceCredentials(in.authParams)
-			if err != nil {
-				in.log.Errorw("not able to get access token", "error", err)
-				return
-			}
-			if err := in.run(); err != nil {
-				in.log.Errorw("got error while running input", "error", err)
-				return
+
+			rt := rate.NewLimiter(rate.Every(retryInterval), 1)
+
+			for in.workerCtx.Err() == nil {
+				// Creating a new channel for cometd input.
+				in.msgCh = make(chan bay.MaybeMsg, 1)
```

Comment:
Can you please re-check there exist a proper cleanup of channel if there exist a situation of continuous error?

Reply:
Hey @agithomas,
Yes, we have thoroughly tested scenarios like continuous errors by following negative test cases.

> [CometD]: Verify by collecting the data for enabled channels by disabling the network for over 1 day, starting data collection and then enabling network
[CometD]: Verify by collecting the data for enabled channels by disabling the network for over 3 hours and re-enable it while data collection is going on
[CometD]: Verify by collecting the data for enabled channels by disabling the network before starting the data collection
[CometD]: Verify by collecting the data for enabled channels by enabling again the network on the system while data collection is in progress

We were not able to re-produce EOF error from salesforce live instance. so we have tested that as well with mock server.

We will share you the list of test cases we have run through in the Slack.

For more detail,
as you can see, I have added child context from worker context which will be helping bayeux for closing channels.
it will not re-create channel until and unless new iteration of input is started.
and for bayeux side, channel closing is properly working. I have verified that as well.

Please let me know if you have any other doubts.